shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/execute/PipelineJobExecutor.java

```diff
@@ -105,6 +103,16 @@ private void execute(final JobConfigurationPOJO jobConfigPOJO) {
         }
     }
     
+    /**
+     * stop job.
+     * @param jobId job id
+     */
+    public void stopJob(final String jobId) {
+        log.info("remove and stop {}", jobId);
+        EXECUTING_JOBS.remove(jobId);
+        RuleAlteredJobSchedulerCenter.stop(jobId);
+    }
```

Comment:
Looks `EXECUTING_JOBS` is not needed any more, could we just depend on `RuleAlteredJobSchedulerCenter.stop` and remove this new method.